locallib.php

```diff
@@ -1863,50 +1867,71 @@ protected function process_edit_actions() {
                 }
                 $openlinkinnewwindow = optional_param('openlinkinnewwindow', false, PARAM_BOOL);
                 $this->updateitem($itemid, $displaytext, $duetime, $linkcourseid, $linkurl, $groupingid, $openlinkinnewwindow);
+                $triggerupdate = true;
                 break;
             case 'deleteitem':
                 if (($this->checklist->autopopulate) && (isset($this->items[$itemid])) && ($this->items[$itemid]->moduleid)) {
                     $this->toggledisableitem($itemid);
                 } else {
                     $this->deleteitem($itemid);
                 }
+                $triggerupdate = true;
                 break;
             case 'moveitemup':
                 $this->moveitemup($itemid);
+                $triggerupdate = true;
                 break;
             case 'moveitemdown':
                 $this->moveitemdown($itemid);
+                $triggerupdate = true;
                 break;
             case 'indentitem':
                 $this->indentitem($itemid);
+                $triggerupdate = true;
                 break;
             case 'unindentitem':
                 $this->unindentitem($itemid);
+                $triggerupdate = true;
                 break;
             case 'makeoptional':
                 $this->makeoptional($itemid, true);
+                $triggerupdate = true;
                 break;
             case 'makerequired':
                 $this->makeoptional($itemid, false);
+                $triggerupdate = true;
                 break;
             case 'makeheading':
                 $this->makeoptional($itemid, true, true);
+                $triggerupdate = true;
                 break;
             case 'nextcolour':
                 $this->nextcolour($itemid);
+                $triggerupdate = true;
                 break;
 
             case 'showhideitems':
                 $itemids = optional_param_array('items', array(), PARAM_INT);
                 foreach ($itemids as $itemid) {
                     $this->toggledisableitem($itemid);
                 }
+                $triggerupdate = true;
                 break;
 
             default:
                 throw new moodle_exception('invalidaction', 'mod_checklist', '', $action);
         }
 
+        if ($triggerupdate) {
+            $params = array(
+                'contextid' => $this->context->id,
+                'objectid' => $this->checklist->id,
+                'userid' => $USER->id,
```

Comment:
userid should be filled in automatically from $USER->id by ::create(), if none is provided? (I've not double-checked)